p8e-api/src/main/kotlin/io/provenance/engine/service/DataAccessChaincodeService.kt

```diff
@@ -0,0 +1,131 @@
+package io.provenance.engine.service
+
+import cosmos.bank.v1beta1.Tx
+import cosmos.base.abci.v1beta1.Abci
+import cosmos.base.v1beta1.CoinOuterClass
+import cosmos.tx.v1beta1.ServiceOuterClass
+import io.p8e.crypto.Hash
+import io.p8e.util.*
+import io.provenance.engine.crypto.toSignerMeta
+import io.provenance.engine.config.ChaincodeProperties
+import io.provenance.engine.crypto.Account
+import io.provenance.engine.crypto.Bech32
+import io.provenance.engine.crypto.toBech32Data
+import io.provenance.p8e.shared.extension.logger
+import io.provenance.metadata.v1.*
+import io.provenance.p8e.shared.service.AffiliateService
+import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPublicKey
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.stereotype.Service
+import p8e.Jobs
+import java.security.KeyPair
+import java.security.PublicKey
+
+@Service
+class DataAccessChaincodeService(
+    private val chaincodeProperties: ChaincodeProperties,
+    private val p8eAccount: Account,
+    private val provenanceGrpcService: ProvenanceGrpcService,
+    private val affiliateService: AffiliateService,
+    private val chaincodeInvokeService: ChaincodeInvokeService,
+) : JobHandlerService {
+    private val log = logger()
+
+    override fun handle(payload: Jobs.P8eJob) {
+        val msgAddScopeDataAccessRequest = MsgAddScopeDataAccessRequest.newBuilder().addAllDataAccess(payload.msgAddScopeDataAccessRequest.dataAccessList)
+            .addAllSigners(payload.msgAddScopeDataAccessRequest.signersList)
+            .setScopeId(payload.msgAddScopeDataAccessRequest.scopeId)
+            .build()
+        val publicKey = payload.msgAddScopeDataAccessRequest.publicKey.toPublicKey()
+
+        val affiliate =
+            transaction { affiliateService.get(publicKey) }.orThrowNotFound("Affiliate with public key ${publicKey.toHex()} not found")
+        val affiliateKeyPair =
+            KeyPair(affiliate.publicKey.value.toJavaPublicKey(), affiliate.privateKey.toJavaPrivateKey())
```

Comment:
We may need to add SmartKey support here as well ... probably setup data under KeyRef data class (hopefully this branch goes out before mine so you don't have to worry about it) :) 

Essentially instead of a KeyPair, we have KeyRef that contains key information such as public key, maybe private key?, UUID (SmartKey way of identifying a private key stored on their system).